### DISCUSSION TITLE
NEEY 4A Smart Active Balancer

### DISCUSSION THREAD

**Discussion by cristi2005** [syssi/esphome-jk-bms#106] [General]
Hi,
I want to buy a NEEY balancer, can this balancer work with this  implementation via bluetooth ?
This balancer: https://www.aliexpress.com/item/1005003125644284.html?spm=a2g0o.cart.0.0.49e33c00nrZQt4&mp=1
BR. Cristi

**Comment by syssi** [comment]
I've disassembled the android app `NEEY_V3.1_2021-11-01.apk`. The class `jy.com.lali.entities.BleMessage` reveals the start of frame `0x55 0xAA 0x11` and an end of frame `0xFF`. This is different to the JK-BMS (SOF: `0x55 0xAA 0xEB` and EOF: Checksum). In other words: This implementation doesn't support the NEEY balancer yet.

**Reply by cristi2005** [reply]
I need a 4Ah balancer, this balancer work with this integration ?
https://www.aliexpress.com/item/1005003423093531.html?spm=a2g0o.cart.0.0.49e33c00nrZQt4&mp=1&gatewayAdapt=glo2deu
Thank you for you response.

**Reply by syssi** [reply]
I've disassembled the Heltec BMS app from https://heltec-bms.com/wp-content/uploads/2022/03/heltec_v2.8_Rel2.zip. It's a copy of the NEEY app above. If you diff the BleMessage of both apps there is no difference:

```
$ diff -u NeeyBleMessage.java HeltecBleMessage.java 
--- NeeyBleMessage.java 2022-05-11 10:58:12.628604686 +0200
+++ HeltecBleMessage.java       2022-05-11 10:57:41.931995249 +0200
@@ -1,6 +1,6 @@
 package jy.com.lali.entities;
 
-import com.uniondevice.jwneey.R;
+import com.uniondevice.jwheltec.R;
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 import java.util.ArrayList;
```

It looks like the NEEY product is a relabeled Heltec device or vice versa.

**Reply by cristi2005** [reply]
Thank you Sebastian for the info.

Have a good day

**Reply by Velociraptor85** [reply]
@syssi is there an idea to extend or fork this project to support these BMSes?

**Reply by syssi** [reply]
@Velociraptor85 The device (4A balancer) is supported since a few days: https://github.com/syssi/esphome-jk-bms/blob/main/esp32-heltec-balancer-ble-example.yaml

**Comment by jhancock4d** [comment]
I cannot for the life of me find the mac for the neey balancer.  Anyone have a the first few characters so that I can narrow down what it should be?

**Reply by syssi** [reply]
One little but important detail: Please note down the MAC address (shown at the app) and force close the app. The balancer can handle a single connection only and stops BLE advertising as soon an active connection is established. So please make sure the android app isn't connected if the ESP32 tries to establish a connection.

**Reply by stefanK74** [reply]
Powerup the balancer, install the Neey app, enable bluetooth and locations, open the app, it scans for available devices, connect, there you go on the app´s landing page..
